File: src/main/java/cz/vhromada/catalog/mapper/ProgramMapper.java
package cz.vhromada.catalog.mapper;

import cz.vhromada.catalog.entity.Program;

import org.mapstruct.Mapper;

/**
 * An interface represents mapper for program.
 *
 * @author Vladimir Hromada
 */
@Mapper
public interface ProgramMapper {

    /**
     * Maps entity program to domain program.
     *
     * @param source entity program
     * @return mapped domain program
     */
    cz.vhromada.catalog.domain.Program map(Program source);

    /**
     * Maps domain program to entity program.
     *
     * @param source domain program
     * @return mapped entity program
     */
    Program mapBack(cz.vhromada.catalog.domain.Program source);

}

File: src/main/kotlin/cz/vhromada/catalog/mapper/ProgramMapper.kt
package cz.vhromada.catalog.mapper

import cz.vhromada.catalog.entity.Program
import cz.vhromada.common.mapper.Mapper
import org.springframework.stereotype.Component

/**
 * A class represents mapper for program.
 *
 * @author Vladimir Hromada
 */
@Component("programMapper")
class ProgramMapper : Mapper<Program, cz.vhromada.catalog.domain.Program> {

    override fun map(source: Program): cz.vhromada.catalog.domain.Program {
        return cz.vhromada.catalog.domain.Program(
                id = source.id,
                name = source.name!!,
                wikiEn = source.wikiEn,
                wikiCz = source.wikiCz,
                mediaCount = source.mediaCount!!,
                crack = source.crack!!,
                serialKey = source.serialKey!!,
                otherData = source.otherData,
                note = source.note,
                position = source.position)
    }

    override fun mapBack(source: cz.vhromada.catalog.domain.Program): Program {
        return Program(
                id = source.id,
                name = source.name,
                wikiEn = source.wikiEn,
                wikiCz = source.wikiCz,
                mediaCount = source.mediaCount,
                crack = source.crack,
                serialKey = source.serialKey,
                otherData = source.otherData,
                note = source.note,
                position = source.position)
    }

}
